### ISSUE TITLE
RV1 please justify the benefit of this approach

### ISSUE THREAD

**Issue by wanted2** [cybercore-co-ltd/CoLAD#1]
Thanks all for the interesting work!

I just have a small concern and hope to get through!

**Can you justify the benefit of this approach?**

I am dissapointed that there is no portion in the paper showing the gain in term of model size and inference speed. Futhermore, when seeing the accuracy of small models like Resnet18 or Resnet 50, they look like inferior results. Because the best accuracy in this paper does not get into top results in COCO leaderboard, I assume that justifications in one of the following aspects (or both) are needed.

- model size
- inference speed

Sorry if there is any misunderstading!

The expectation is that the authors can justify the benefit among many other approaches which also reduce model size and increase speed, such as model pruning, NAS, or model compression methods like quantization, ...

**Comment by chuong98** [comment]
Hi, thanks for interesting in our paper.

> there is no portion in the paper showing the gain in term of model size and inference speed.

Please let me clarify a potentially misunderstanding in here:
 + Distillation aims to **improve model accuracy but does not modify the network architecture**. Therefore, the model size and speed remain the same, but the accuracy improves. For example, we show that PAA detector with ResNet50 backbone can match the performance of that using ResNet101, which is much larger backbone. 

+ What you refer, e.g. model pruning, NAS, quantization, is about Network Compression technique. That is, we hope to **maintain the model accuracy while reducing the memory and improving inference speed**.  

+ Distillation and Network Compression may be often used together, for example, after being pruned/quantized, the acuuracy of model drops, and we can use Distillation techniques to regain the accuracy. However, they are two **completely different concepts**, trying achieve **different goals**.

**Comment by chuong98** [comment]
>the accuracy of small models like Resnet18 or Resnet 50, they look like inferior results.

No, the models have been **significantly improved** using our LAD methods, please refer to Tab 11 in the Appendix B for comparing the small models. 

<img src="https://user-images.githubusercontent.com/15174756/131552082-6d282d91-cd17-4691-a0e7-659b0ca71209.png " height="200">

Specifically, our proposed techniques are **LAD/CoLAD/SoLAD**, and we show that:

+ For ResNet18 backbone, SoLAD improves the PAA baseline from 35.8AP to 38AP **(+2.2)** and 38.4AP **(+2.6)** using PAA-R50 and PAA-R101 teacher, respectively. The improvement is much higher than the LD method, which only gains +0.1 and +0.7, respectively. 

+ For ResNet50, we show that we can improve the baseline PAA-R50 from 40.4AP to 42.4AP(+2AP).

**Comment by chuong98** [comment]
>The best accuracy in this paper does not get into top results in COCO leaderboard

Please note that top results on COCO Leader Board are achieved with very large backbone, much longer training (x6 -x12 schedule), using Test-Time Augmentation, extra data, Auto-Augmentation, and extremely heavy head, like Cascaded-Mask-RCNN or Hybrid Task Cascade or add-on attention blocks. These models are great for some competitions, but rarely for practical usage due to computation cost. 

Unfortunately, we don't have the free resource now to train such models. Our objective is rather to improve the accuracy of standard detectors without bell-and-whistle. 

Note that, for PAA-R50 backbone,
+ we can improve PAA-R50 to 43.7AP with 1x training schedule (12 epochs). To the best of our knowledge, we are not aware of any detectors/methods that can achieve this high accuracy training for the same number of iterations/setting/backbone. Please help us to correct if we miss any good methods, and we will update the paper accordingly.
+ In contrast, using the default PAA-R50 you need to train 3x schedule (36 epochs) to get to 43.8AP.  
+ Longer training not only costs your time/resource, but also overfits the model, hence requiring additional tricks, e.g. extensive augmentation, extra data.

**Comment by chuong98** [comment]
So, I guess to answer your big question " What is the benefit of the our method?", the informal answers will be:
+ Given a model object detection, if you want to improve its performance by 1-1.5mAP, use our CoLAD.
+ If you already have another smaller model pretrained on the target dataset, e.g., COCO, great, then use it as the teacher and train your target model with LAD.
+ If you already have a larger teacher model, awesome, then use SoLAD.  The performance will improve significantly. 
+ If you want the model run faster, then use other techniques, such as Pruning, Quantization.

**Comment by wanted2** [comment]
@chuong98 I don't understand why the authors closed this issue and other issues (RV2,3,4,5).

I'm sorry but I am not persuaded with the benefits the authors gave.
In general, Knowledge Distillation (KD) aims at distilling the knowledge of a `cumbersome` teacher to much `smaller and lightweight` students.
However, in this work, all students are `cumbersome`, so I am confused about if the paper give some contributions to the field.
If the `cumbersome` students are the outcomes, then why one shouldn't take cumbersome non-KD counterparts such as the [SoftTeacher](https://github.com/microsoft/SoftTeacher) is not quite clear.
If the proposed methods can improve `lightweight` students, then the experiments in the paper is insufficient.

So, what is the main problem and the key idea to solve these challenges? What is the main contribution?

Notes:
* _The authors did some experiments for EfficientNet but these details are not in the paper body._
* The experiments of EfficientNet in appendices are for an ablation study which does not focus on the main contribution.

**Comment by chuong98** [comment]
>I don't understand why the authors closed this issue and other issues (RV2,3,4,5).

@wanted2 Please feel free to reopen any issues that you may want, we will try our best to clarify/ answer your questions.

> Knowledge Distillation (KD) aims at distilling the knowledge of a cumbersome teacher to much smaller and lightweight students. 

- It depends on how you define `cumbersome`. The convention is that you choose a target network, and use another network (teacher) to distill. With that said, Teacher and Student can be any thing, depend on your application.
   +  If your goal is to deploy on edge device, then MobileNet backbone can be a favorite one. 
   +  However, if my target device is FPGA or GPU, and I want to have a high accuracy model, well supported operators, then ResNet suits better.   
   +  We choose ResNet18 and ResNet50 simply because they are standard and commonly used for ablation comparison. You can try our method on any backbone that you want.

- Our method can work with any kinds of pair Teacher - Student, even if the Teacher is a smaller one. 
Specifically, we show that Using Teacher PAA-R101, we can improve Student PAA-R50 match the performance of teacher. 
- For smaller Student backbone, such as R18, the results are also significantly improved, see Table 11. 
<img src="https://user-images.githubusercontent.com/15174756/131552082-6d282d91-cd17-4691-a0e7-659b0ca71209.png " height="200">

- If you are still persistent that `MobileNet` backbone must be used, we will release the code when the paper is accepted, then you can train your own model. 
 
Note, we **keep our right to release the checkpoints**, and the released code/results  are **only for academic purpose**. Any commercial usage **are prohibited.** 
We really appreciate any questions that help clarify and improve the paper quality, and that is why we will open the source code to facilitate the reproduction.
However, we reserve our right to refuse any unreasonable, impolite requests, which go against the academic purpose.

**Comment by shinya7y** [comment]
Hi, @chuong98

> we can improve PAA-R50 to 43.7AP with 1x training schedule (12 epochs). To the best of our knowledge, we are not aware of any detectors/methods that can achieve this high accuracy training for the same number of iterations/setting/backbone. Please help us to correct if we miss any good methods, and we will update the paper accordingly.

UniverseNet-20.08 variants achieve 44.7AP (ResNet-50-B) and 45.8AP (ResNet-50-C) on COCO val. https://arxiv.org/abs/2103.14027
I think the paper will be more convincing by training UniverseNet-20.08 without GFL with PAA and CoLAD.

**Comment by chuong98** [comment]
Hi @shinya7y , thanks for introducing UniverseNet, the results are very impressed !!!
>UniverseNet-20.08 variants achieve 44.7AP (ResNet-50-B) and 45.8AP (ResNet-50-C) on COCO val. >https://arxiv.org/abs/2103.14027

For exact reproduction, can you please tell us the config files (link) and (ckpt if any) for the model USB-R50-B/C mentioned above.
We would like clarify the setting details, and we will try the USBNet with PAA/CoLAD accordingly.
Thank you.
